src/backends/ohos.rs

```diff
@@ -0,0 +1,166 @@
+    }
+
+    fn buffer_mut(&mut self) -> Result<BufferImpl<'_, D, W>, SoftBufferError> {
+        let native_window_buffer = self.native_window.request_buffer(None).map_err(|err| {
+            SoftBufferError::PlatformError(
+                Some("Failed to request native window buffer".to_owned()),
+                Some(Box::new(err)),
+            )
+        })?;
+
+        if !matches!(
+            native_window_buffer.format(),
+            // These are the only formats we support
+            NativeBufferFormat::RGBA_8888 | NativeBufferFormat::RGBX_8888
+        ) {
+            return Err(SoftBufferError::PlatformError(
+                Some(format!(
+                    "Unexpected buffer format {:?}, please call \
+                    .resize() first to change it to RGBx8888",
+                    native_window_buffer.format()
+                )),
+                None,
+            ));
+        }
+        let size = (native_window_buffer.width() * native_window_buffer.height())
+            .try_into()
+            .map_err(|e| {
+                SoftBufferError::PlatformError(
+                    Some("Failed to convert width to u32".to_owned()),
+                    Some(Box::new(e)),
+                )
+            })?;
+        let buffer = vec![0; size];
+
+        Ok(BufferImpl {
+            native_window_buffer,
+            buffer,
+            marker: PhantomData,
+        })
+    }
+
+    /// Fetch the buffer from the window.
+    fn fetch(&mut self) -> Result<Vec<u32>, SoftBufferError> {
+        Err(SoftBufferError::Unimplemented)
+    }
+}
+
+pub struct BufferImpl<'a, D: ?Sized, W> {
+    native_window_buffer: NativeWindowBuffer<'a>,
+    buffer: Vec<u32>,
+    marker: PhantomData<(&'a D, &'a W)>,
+}
+
+unsafe impl<'a, D, W> Send for BufferImpl<'a, D, W> {}
+
+impl<'a, D: HasDisplayHandle, W: HasWindowHandle> BufferInterface for BufferImpl<'a, D, W> {
+    #[inline]
+    fn pixels(&self) -> &[u32] {
+        &self.buffer
+    }
+
+    #[inline]
+    fn pixels_mut(&mut self) -> &mut [u32] {
+        &mut self.buffer
+    }
+
+    #[inline]
+    fn age(&self) -> u8 {
+        0
+    }
+
+    // TODO: This function is pretty slow this way
+    fn present(mut self) -> Result<(), SoftBufferError> {
+        let input_lines = self.buffer.chunks(self.native_window_buffer.width() as _);
+        for (output, input) in self
+            .native_window_buffer
+            .lines()
+            // Unreachable as we checked before that this is a valid, mappable format
+            .unwrap()
+            .zip(input_lines)
+        {
+            // .lines() removed the stride
+            assert_eq!(output.len(), input.len() * 4);
+
+            for (i, pixel) in input.iter().enumerate() {
+                // Swizzle colors from RGBX to BGR
+                let [b, g, r, _] = pixel.to_le_bytes();
+                output[i * 4].write(b);
+                output[i * 4 + 1].write(g);
+                output[i * 4 + 2].write(r);
```

Comment:
Since this entire module was copied from Android, keep in mind that I recently fixed some typos in this borked swizzle: https://github.com/rust-windowing/softbuffer/commit/ae1565bcdc177aaad6faa7382a1fd4d5f6fde472